src/konnektor/network_tools/misc.py

```diff
@@ -1,37 +1,98 @@
+                              Component, Component]]) -> LigandNetwork:
+    """
+    Remove the desired edge from the network
 
-def delete_transformation(network :LigandNetwork,
-                          edge :tuple[SmallMoleculeComponent, SmallMoleculeComponent])\
-        ->LigandNetwork:
-    if(isinstance(edge, LigandAtomMapping)):
-        edge = (edge.componentA, edge.componentB)
+    Parameters
+    ----------
+    network: LigandNetwork
+    transformation: :Union[LigandAtomMapping, tuple[Component, Component]]
 
-    f = lambda m: len({m.componentA, m.componentB}.union(edge)) != 2
-    filtered_edges = filter(f, network.edges)
+    Returns
+    -------
+    LigandNetwork
+        returns a copy of the ligand network without the removed edge.
+    """
+    if (isinstance(transformation, LigandAtomMapping)):
+        transformation = (transformation.componentA, transformation.componentB)
+
+    f = lambda m: len({m.componentA, m.componentB}.union(transformation)) != 2
+    filtered_edges = list(filter(f, network.edges))
 
     return LigandNetwork(edges=filtered_edges, nodes=network.nodes)
 
 
-def cluster_compound(compounds: Iterable[SmallMoleculeComponent],
-                     featurize:TransformerMixin = RDKitFingerprintTransformer(),
-                    cluster:ClusterMixin = KMeans(n_clusters=5, n_init="auto")
-                     )->dict[int,
-list[SmallMoleculeComponent]]:
-    
-    clusterer = ComponentsDiversityClustering(featurize = featurize, cluster=cluster)
-    return clusterer.cluster_compounds(compounds)
+def delete_component(network: LigandNetwork,
+                     component: Component) -> LigandNetwork:
+    """
+    Remove the desired component, which is a node of the graph, and its
+    edges from the network.
+
+    Parameters
+    ----------
+    network: LigandNetwork
+    component: Component
+        component to be removed from the network, which is a node of the graph.
+
+    Returns
+    -------
+    LigandNetwork
+        returns a copy of the ligand network without the removed component
+         and edges containing the component.
+    """
+    filtered_nodes = list(filter(lambda n: n != component, network.nodes))
+
+    f = lambda m: component not in (m.componentA, m.componentB)
+    filtered_edges = list(filter(f, network.edges))
+
+    return LigandNetwork(edges=filtered_edges, nodes=filtered_nodes)
+
+
+def cluster_components(compounds: Iterable[Component],
+                       clusterer: _AbstractClusterer
+                       ) -> dict[int, list[Component]]:
+    """
+    This is a helper function for using the clustering alsgorithms.
+    TODO: Implement this
+    Not Implemented!
+
+    Parameters
+    ----------
+    compounds: Iterable[SmallMoleculeComponent]
+    clusterer: _AbstractClusterer
+        the function used to featurize the
+
+    Returns
+    -------
+
+    """
+    raise NotImplementedError()
```

Comment:
isn't this just `clusterer.cluster_compounds(compounds)`?  I.e. it's a method on the class not a function separate from the clusterer class

Reply:
yeah, I was considering to kick it again. I will do it.